nodes/vector/vector_dot_product.py

```diff
@@ -0,0 +1,14 @@
+import bpy
+from ... base_types.node import AnimationNode
+
+class VectorDotProductNode(bpy.types.Node, AnimationNode):
```

Comment:
Should we maybe make this operation part of the vector math node?
the api seems to be wrong about this operation. It sais the dot function returns a vector: http://www.blender.org/api/blender_python_api_2_76_1/mathutils.html#mathutils.Vector.dot